### PR TITLE
Register udfs against the pg_catalog schema.

### PR DESCRIPTION
**Pull request by andreidan**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by seut** [comment]
Do we really want to allow that? If we do, why only pg_catalog and no other internal schemas?
@mfussenegger What is your opinion here?

**Comment by andreidan** [comment]
I'd say this is useful to improve compatibility with various tools/clients without having to do a cratedb release to implement missing `pg_catalog` stuff

**Comment by andreidan** [comment]
Postgres seems to allow the creation of udfs against any schema

**Comment by seut** [comment]
Yep so we should also allow any or none. Otherwise it will be hard to document why its allowed on this schema but not on that.

**Comment by mfussenegger** [comment]
I think it is fine to open this up for `pg_catalog` for compatibility reasons.

Not so sure about `sys` or `information_schema`. Afaik the compatibility argument doesn't hold for those - and I'd rather prevent users from creating UDFs inside those schemas. It breaks consistency, but I think I'm in favour of treating `pg_catalog` special instead of generally opening it up.

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -222,6 +222,9 @@ Changes
 Fixes
 =====
 
+- Allow :ref:`sql_administration_udf` to be registered against the
```
Comment (seut) [inline blackbox/docs/appendices/release-notes/unreleased.rst]: This is a feature not a fix.

**Pull request merged**